app/src/main/java/com/github/multimatum_team/multimatum/activity/DeadlineDetailsActivity.kt

```diff
@@ -58,10 +59,19 @@ class DeadlineDetailsActivity : AppCompatActivity() {
     private lateinit var descriptionView: EditText
 
     // Set them on default value, waiting the fetch of the deadlines
-    private var notificationSelected: BooleanArray = booleanArrayOf(false, false, false, false)
     private var dateTime: LocalDateTime = LocalDateTime.of(2022, 10, 10, 10, 10)
     private var state: DeadlineState = DeadlineState.TODO
 
+    // Memorisation of which checkBox is selected for the notifications
+    private var notificationSelected: BooleanArray = booleanArrayOf(false, false, false, false)
+    private var nameNotifications = arrayOf("1 hour", "5 hours", "1 day", "3 days")
+    private var timeNotifications = arrayOf(
+        Duration.ofHours(1).toMillis(),
+        Duration.ofHours(5).toMillis(),
+        Duration.ofDays(1).toMillis(),
+        Duration.ofDays(3).toMillis()
+    )
```

Comment:
If I am not mistaken this array is already created in `AddDeadlineActivity`. Do we really need to duplicate it here?

Reply:
We need a duplicate if we want default options when we modify the deadline.  We could put it somewhere else and recuperate it but I think that would make the code less readable, but it is just my opinion.

Reply:
I am not sure of how it works, but from what I see it is duplicate state and that is dangerous. Maybe you can store these values into an immutable list, and call `toArray` (or equivalent, I don't remember exactly) on this list each time you need an array. So you would have the default values in the list and no dependency between the arrays.